src/app/core/http-services/item-navigation.service.ts

```diff
@@ -199,4 +199,41 @@ export class ItemNavigationService {
     return (isSkill(type)) ? this.getRootSkills() : this.getRootActivities();
   }
 
+  getNavigationNeighbors(itemRoute: ItemRoute): Observable<{parent: ItemRoute|null, left: ItemRoute|null, right: ItemRoute|null}> {
+    if (!isRouteWithAttempt(itemRoute)) return EMPTY;
+
+    // Root activity => no parent/left/right activity
+    if (itemRoute.path.length === 0) return of({ parent: null, left: null, right: null });
+
+    const parentId = itemRoute.path[itemRoute.path.length - 1] as string;
+
+    return this.getNavDataFromChildRoute(parentId, itemRoute).pipe(map(navParent => {
+      const index = navParent.items.findIndex(item => item.id === itemRoute.id);
+
+      if (index === -1) throw new Error(`Unexpected: item is missing from its parent children list`);
+
+      const left: ItemRoute|null = index === 0 ? null : {
+        id: navParent.items[index - 1]?.id as string,
+        contentType: 'activity',
+        attemptId: itemRoute.attemptId,
```

Comment:
I didn't quite understand which attempt I should take for the siblings...
From my understanding, I should set `attemptId` to the `attemptId` of the sibling if it's not null, and set `parentAttemptId` to the `attemptId` of its parent otherwise?
And for the parent, just set the `attemptId` to the `attemptId` of the current Item or its `parentAttemptId`

Reply:
correct for the siblings, incorrect for the parent. See my comments in the next review.